hack/jenkins/e2e.sh

```diff
@@ -80,8 +103,14 @@ fi
 # Have cmd/e2e run by goe2e.sh generate JUnit report in ${WORKSPACE}/junit*.xml
 export E2E_REPORT_DIR=${WORKSPACE}
 
+function teardown() {
+  echo "Cleaning up with e2e --down"
+  go run ./hack/e2e.go ${E2E_OPT} -v --down
+}
+
+trap "teardown" EXIT
```

Comment:
Yeah, take this out. We decided to go with "tests never cause a non-zero exit code", which is the Jenkins way.